### PR TITLE
Add explicit mention of exotic energy hatches

### PR DESCRIPTION
**Pull request by FourIsTheNumber**
Many machines still take TecTech Multi-Amp/Laser hatches without mentioning this critical information in their tooltips. This adds a .addTecTechHatch line to the tooltip builder that displays this line:
image
This also adds the line to every multiblock I am aware of that accepts these hatches and replaces existing mentions with this line. Notably, it has been explicitly added to every TecTech multiblock - the player should not have to assume which multiblocks can use the hatches based on the mod (more progress towards mergening).

Additionally, added the line: 
image
to all megas.

### CODE REVIEW

File: src/main/java/gregtech/api/util/MultiblockTooltipBuilder.java [hunk 1]
```diff
@@ -546,6 +547,17 @@ public MultiblockTooltipBuilder addOtherStructurePart(String name, String info,
         return this;
     }
 
+    /**
+     * Add a line of information about the structure:<br>
+     * Supports TecTech Multi-Amp and Laser Hatches!
+     *
+     * @return Instance this method was called on.
+     */
+    public MultiblockTooltipBuilder addTecTechHatch() {
```
Comment (wlhlm) [inline src/main/java/gregtech/api/util/MultiblockTooltipBuilder.java:556]: I'd prefer calling this `addTecTechHatchInfo` in line with the `addInfo` method.

**Review by wlhlm — CHANGES_REQUESTED** [review]
Looks good minus the method name bikeshed.

**Comment by wlhlm** [comment]
> Additionally, added the line:
> > Processes up to 256 recipes at once
> 
> to all megas.

Could this be programmatically generated rather than hardcoded?

**Comment by FourIsTheNumber** [comment]
> Could this be programmatically generated rather than hardcoded?

There's no mega base class (and of course there are only 6). I could have this call getMaxParallel instead of hardcoding 256, though.

File: src/main/java/bartworks/common/tileentities/multis/mega/MTEMegaBlastFurnace.java [hunk 2]
```diff
@@ -172,11 +172,13 @@ public IMetaTileEntity newMetaEntity(IGregTechTileEntity iGregTechTileEntity) {
     protected MultiblockTooltipBuilder createTooltip() {
         MultiblockTooltipBuilder tt = new MultiblockTooltipBuilder();
         tt.addMachineType("Blast Furnace")
+            .addInfo("Processes up to " + Configuration.Multiblocks.megaMachinesMax + " recipes at once")
```
Comment (wlhlm) [inline src/main/java/bartworks/common/tileentities/multis/mega/MTEMegaBlastFurnace.java:175]: Can you wrap this in a `addParallelInfo` or something like that?

**Review by wlhlm — CHANGES_REQUESTED** [review]
Nice!

File: src/main/java/bartworks/common/tileentities/multis/mega/MTEMegaBlastFurnace.java (hunk 2 above)
Comment (FourIsTheNumber) [reply src/main/java/bartworks/common/tileentities/multis/mega/MTEMegaBlastFurnace.java]: Added - I feel a little weird adding this here and not back-adding it to other multis, so I'll probably do a follow-up pr to add standardized tooltips for things like parallels per tier, speed bonus, eu usage, etc.

Comment (wlhlm) [reply src/main/java/bartworks/common/tileentities/multis/mega/MTEMegaBlastFurnace.java]: > so I'll probably do a follow-up pr to add standardized tooltips for things like parallels per tier, speed bonus, eu usage, etc.

Awesome!

**Review by wlhlm — APPROVED** [review]

**Pull request merged**